### PR TITLE
test: Allow for asnychronous DOM changes in storagelib

### PR DESCRIPTION
**Pull request by mvollmer**
The code was careful not to click on non-existing elements, for
example, but it assumed a stable DOM so that there are no
check-before-use races.  This was true in the days of PhantomJS but is
no longer true with CDP.

Thus, we now allow a testlib.Error and retry.

### CODE REVIEW

File: test/verify/storagelib.py [hunk 1]
```diff
@@ -167,29 +167,42 @@ def check():
...
-            if not b.is_present(cell) or not b.is_visible(cell):
+                if val is not None and val in b.text(cell):
+                    return True
+                if alternate_val is not None and alternate_val in b.text(cell):
+                    return True
+                if cond is not None and cond(cell):
+                    return True
+                return False
+            except Error as ex:
+                print("RETRY", ex)
```
Comment (marusak) [inline test/verify/storagelib.py:204]: In the first run this storage test did not fail even once, but nor this RETRY was ever caught, lets try again. Force triggering it all.

Comment (marusak) [reply test/verify/storagelib.py:204]: I could reproduce it yesterday, so I'll try it locally and see, if I can hit that RETRY.

Comment (marusak) [reply test/verify/storagelib.py:204]: ok, I could reproduce it locally twice, once with
```
RETRY eval_js(ph_is_visible("#detail-content > article > div > table > tbody:nth-of-type(1) .ct-listing-panel-body:nth-child(2) dt:contains(Name) + *")): #detail-content > article > div > table > tbody:nth-of-type(1) .ct-listing-panel-body:nth-child(2) dt:contains(Name) + * not found
```
and once with
```
RETRY timeout
wait_js_cond(ph_is_visible("#detail-content > article > div > table > tbody:nth-of-type(1) .ct-listing-panel-body:nth-child(2) dt:contains(Name) + *")): {'exceptionId': 1, 'text': 'Uncaught (in promise) Error: condition did not become true', 'lineNumber': 233, 'columnNumber': 9, 'scriptId': '3', 'stackTrace': {'callFrames': [{'functionName': 'PhWaitCondTimeout', 'scriptId': '3', 'url': '', 'lineNumber': 232, 'columnNumber': 8}, {'functionName': '', 'scriptId': '3', 'url': '', 'lineNumber': 244, 'columnNumber': 23}]}, 'exception': {'type': 'object', 'subtype': 'error', 'className': 'PhWaitCondTimeout', 'description': 'Error: condition did not become true\n    at <anonymous>:245:24', 'objectId': '{"injectedScriptId":15,"id":2}'}}
cdp: {'source': 'network', 'level': 'error', 'text': 'Failed to load resource: the server responded with a status of 401 (Authentication failed)', 'timestamp': 1611315578506.5962, 'url': 'http://127.0.0.2:9091/cockpit/login', 'networkRequestId': '3436487.12'}
```
and in both cases the test eventually succeeded so it seems to be working.

Comment (marusak) [reply test/verify/storagelib.py:204]: Do you want to keep that print? Feels a bit pointless really.

Comment (mvollmer) [reply test/verify/storagelib.py:204]: > ok, I could reproduce it locally twice,

Cool, thanks!

Comment (mvollmer) [reply test/verify/storagelib.py:204]: > Do you want to keep that print? Feels a bit pointless really.

Yes, I agree, let's remove it.

Comment (mvollmer) [reply test/verify/storagelib.py]: > Yes, I agree, let's remove it.

Done.

**Review by marusak — APPROVED** [review]

**Pull request merged**